### ISSUE TITLE
Web - Double <Payer:> for IOU report preview in Search

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#33760]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.19-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. [User A] Request money from User B.
2. [User B] Pay the request.
3. [User A] Click on the IOU preview to open IOU report.
4. [User A] Click Search icon.
5. Note that the IOU report shows double <Payer:> in the preview in Search list.

## Expected Result:
Single <Payer:> for IOU report preview in Search

## Actual Result:
Double <Payer:> for IOU report preview in Search

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![Screenshot 2023-12-29 at 22 37 08](https://github.com/Expensify/App/assets/93399543/7d3fb263-a87c-4033-bf63-80d0884682bb)

https://github.com/Expensify/App/assets/93399543/943bd907-86b0-4617-8b73-b1ccea6ff615

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
Issue is not Reproduced on production

![Screenshot 2023-12-29 at 22 44 08](https://github.com/Expensify/App/assets/93399543/a3621cec-58bc-49ba-b764-111723da63ec)

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


### What is the root cause of that problem?
This is a regression from https://github.com/Expensify/App/issues/30042

We now prepend the payer name twice: 

https://github.com/Expensify/App/blob/8432563c2fc7f54f19d927d457833c594fb3d9d5/src/libs/ReportUtils.ts#L2029-L2032

And:

https://github.com/Expensify/App/blob/9692e48dceeafa9d2151d3e54140295c690c2f73/src/libs/OptionsListUtils.js#L512-L516

### What changes do you think we should make in order to solve the problem?
Since the lastMessageTextFromReport has changed for the money reports to include the payer name, we should handle this case and set `result.alternateText` to only `lastMessageTextFromReport`, not `lastMessageText` for such options.

### What alternative solutions did you explore? (Optional)

**Comment by mountiny** [comment]
I think this was known and wanted to address in a follow up @koko57

**Comment by koko57** [comment]
It was occuring long before the merge and I found it while I was working on it (it was spotted by me on staging) https://github.com/Expensify/App/issues/30042#issuecomment-1864366783

it displayed [firstName lastName]: [firstName]: but after my changes I had to make for LHN it displays firstName doubled. We agreed that it will be fixed in a separate PR

**Comment by mountiny** [comment]
Yep its been listed in the issue already as a follow up, i think that this is not a blocker as it only appears in the search page and we will address this in New Year

**Comment by koko57** [comment]
Will take care of it this week 🙂

**Comment by koko57** [comment]
@mountiny one question - should the previews in the SearchPage be 1:1 similar to LHN previews? If yes I'd use the same logic as in the LHN previews and maybe create a separate function out of it.

 I've spotted another weird thing here - for some of the chats I have a strange fallback message
<img width="371" alt="Screenshot 2024-01-02 at 13 46 31" src="https://github.com/Expensify/App/assets/28020445/d0eaf1fc-b792-4513-b9b7-80c2566ca6fe">
<img width="377" alt="Screenshot 2024-01-02 at 13 51 53" src="https://github.com/Expensify/App/assets/28020445/538fa3b2-5838-4fbd-91aa-c67239a0786e">
<img width="1046" alt="Screenshot 2024-01-02 at 13 52 08" src="https://github.com/Expensify/App/assets/28020445/d8044618-c494-4fc0-89e7-c06abe5bb895">

**Comment by mountiny** [comment]
Yeah i think we can make it 1:1 to the lhn if possible

**Comment by koko57** [comment]
@mountiny I started working on a refactor but it would require a lot of changes. I want to move the logic for creating alternative text out of the `getOptionData` from SidebarUtils and use the same logic in `createOption` OptionListUtils.
But as I wanted to avoid potential regressions and the problems we had with the first PR for LHN previews, maybe I could fix it more simply? 

[I copied the part of logic ](https://github.com/Expensify/App/compare/main...koko57:App:fix/33760-doubled-displayName-on-searchPage?expand=1) that displays the lastActorDisplayName only for certain types of the reportAction and pasted it in the `createOption` function. It's less invasive - so less regression-prone. 
We could fix it this way now and then I could work on the refactor. WDYT? 🙂

**Comment by mountiny** [comment]
I think that ideally we do abstract away code to avoid duplication, but if you feel like this is not worth at this point, we could go this way too for now and refactor once more changes will be required

**Comment by koko57** [comment]
@mountiny I think it is not necessary to extract this logic now, I'd rather do this in this major refactor and extracting all the logic for getting `alternateText` later 🙂

**Comment by mountiny** [comment]
Ok then!

**Comment by koko57** [comment]
PR opened: https://github.com/Expensify/App/pull/33966

**Comment by trjExpensify** [comment]
>I've spotted another weird thing here - for some of the chats I have a strange fallback message
> <img width="1046" alt="Screenshot 2024-01-02 at 13 52 08" src="https://github.com/Expensify/App/assets/28020445/d8044618-c494-4fc0-89e7-c06abe5bb895">

On this, that's the workspace chat, which in search always displays either the word `Workspace` for the member or the `workspaceName` for other participants of the chat (i.e the admin). 

<img width="376" alt="image" src="https://github.com/Expensify/App/assets/16232057/17879208-34fd-4112-abd8-4f16ab496f91">

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/16232057/cc85cff0-59b1-4299-a566-fa5c99a6f6b5">

We shouldn't change that. 👍

**Comment by mountiny** [comment]
@koko57 maybe we dont need that fix then

**Comment by koko57** [comment]
@trjExpensify @mountiny no worries, I didn't change that 🙂

**Comment by mountiny** [comment]
ok cool, thanks for confirming

**Comment by mountiny** [comment]
PR merged

**Comment by mountiny** [comment]
@situchan Can you complete the checklist here? this was a simple follow up to previou PR, I feel like the reward should be $250 for this one

**Comment by situchan** [comment]
BZ Checklist:
Offending PR: https://github.com/Expensify/App/pull/33346, though the root cause already existed - https://github.com/Expensify/App/issues/33760#issuecomment-1872178470
As this was caught during staging regression testing, no need new regression test.
To prevent such bugs further, we decided to make LHN and search logic consistent as follow-up [here](https://github.com/Expensify/App/issues/34411)

**Comment by mountiny** [comment]
> I'm fine with any reward change but does that mean that now we have the right to request bounty increase for reviewing large PRs? For me, some PRs took several days to complete review. I thought they were flatten no matter small or large PR, as they're balanced eventually.

> PS: https://github.com/Expensify/App/pull/33371 was just console log change PR and fully rewarded https://github.com/Expensify/App/issues/33478#issuecomment-1867947159

Yeah I see your point, lets keep it $500 then. 

I think the root cause is that we should get the same C+ review the clean up PRs too as in this case it was a regression from PR reviewed by another C+ so then we end up paying more in total for one task

**Comment by michaelhaxhiu** [comment]
>Yeah I see your point, lets keep it $500 then.

Sounds like we arrived at $500 standard rate for this PR review! I will make an upwork job now for it.

>I thought they were flatten no matter small or large PR, as they're balanced eventually.

FWIW -- what you described here is indeed the core idea at play for the bounty methodology. It's a pay rate that is intended to be more than fair in the grand scheme of being a C+, unless a specific PR is a HUGE deviation from the typical workload and deserves a one-off consideration (e.g. a particular.

**Comment by michaelhaxhiu** [comment]
I just sent you an offer actually. But that's a good reminder of how to use the automation

**Comment by michaelhaxhiu** [comment]
paid
